tests/check/check_userset.go

```diff
@@ -0,0 +1,1000 @@
+				Expectation: false,
+			},
+			{
+				Name:        "valid_user_direct_wild",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:nou_3", Relation: "nested_or_userset", User: "user:nou_3"},
+				Expectation: true,
+			},
+			{
+				Name:        "invalid_user_direct",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:nou_1", Relation: "nested_or_userset", User: "user:nou_2"},
+				Expectation: false,
+			},
+			{
+				Name:        "invalid_user_direct_cond",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:nou_2", Relation: "nested_or_userset", User: "user:nou_1"},
+				Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("1")}},
+				Expectation: false,
+			},
+			{
+				Name:      "user_direct_cond_no_condition",
+				Tuple:     &openfgav1.TupleKey{Object: "usersets-user:nou_2", Relation: "nested_or_userset", User: "user:nou_2"},
+				ErrorCode: 2000,
+			},
+			{
+				Name:        "valid_user_butnot_computed",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:nou_4", Relation: "nested_or_userset", User: "user:nou_4"},
+				Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("1")}},
+				Expectation: true,
+			},
+
+			{
+				Name:        "valid_user_butnot_computed_invalid_cond",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:nou_4", Relation: "nested_or_userset", User: "user:nou_4"},
+				Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("2")}},
+				Expectation: false,
+			},
+			{
+				Name:        "invalid_user_butnot_computed",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:nou_4", Relation: "nested_or_userset", User: "user:nou_5"},
+				Context:     &structpb.Struct{Fields: map[string]*structpb.Value{"x": structpb.NewStringValue("1")}},
+				Expectation: true,
+			},
+			{
+				Name:      "butnot_computed_no_condition",
+				Tuple:     &openfgav1.TupleKey{Object: "usersets-user:nou_4", Relation: "nested_or_userset", User: "user:nou_4"},
+				ErrorCode: 2000,
+			},
+		},
+	},
+	{
+		Name: "usersets_nested_and_userset", // TODO: more cases to be covered?
+		Tuples: []*openfgav1.TupleKey{
+
+			{Object: "directs-user:nau_1", Relation: "direct_cond", User: "user:nau_1", Condition: &openfgav1.RelationshipCondition{Name: "xcond"}},
+			{Object: "directs-user:nau_1", Relation: "direct_wild", User: "user:*"},
+
+			{Object: "usersets-user:nau_1", Relation: "userset_to_and_computed", User: "directs-user:nau_1#and_computed"},
+			{Object: "usersets-user:nau_1", Relation: "userset_to_or_computed", User: "directs-user:nau_1#or_computed"},
+		},
+		CheckAssertions: []*checktest.Assertion{
+			{
+				Name:        "valid_user",
+				Tuple:       &openfgav1.TupleKey{Object: "usersets-user:nau_1", Relation: "userset_to_and_computed", User: "user:nau_1"},
```

Comment:
Why are we testing `userset_to_and_computed` while the name of the test is `usersets_nested_and_userset`?  Should we test `nested_and_userset`?

Reply:
Copy paste error, overlooked that this was computed and just took the last relation in the tuples, fixed in 39f3b5a315b18310a7ec5120f92585299ef3f4fe